core/domain/suggestion_services.py

```diff
@@ -2012,32 +2019,38 @@ def get_all_contributor_stats(
 
 def _update_translation_contribution_stats_models(
     translation_contribution_stats: List[
-        suggestion_registry.TranslationContributionStats]) -> None:
+        suggestion_registry.TranslationContributionStats]
+) -> None:
     """Updates TranslationContributionStatsModel models for given translation
     contribution stats.
 
     Args:
         translation_contribution_stats: list(TranslationContributionStats).
             A list of TranslationContributionStats domain objects.
     """
-    stats_ids = []
     stats_dict = {}
 
     for stat in translation_contribution_stats:
-        # Ruling out the possibility of None for mypy type checking.
-        assert stat is not None
-        stat_id = suggestion_models.TranslationContributionStatsModel.build_id(
-            stat.language_code, stat.contributor_user_id, stat.topic_id) # type: ignore[arg-type]
-        stats_ids.append(stat_id)
+        stat_id = (
+            suggestion_models.TranslationContributionStatsModel.construct_id(
+                str(stat.language_code),
```

Comment:
Why do you need to do all these casts to strings? /cc @sahiljoster32

Reply:
Here, `stat.language_code`, `stat.contributor_user_id` and `stat.topic_id` is of type Optional[str]. Because while creating default TranslationContributionStats domain object we are providing them as None https://github.com/oppia/oppia/blob/7ad815df8c289a032080cc909a03f4a3b37287a6/core/domain/suggestion_registry.py#L1704

So, instead of just casting to string please throw an error, if these values are not expected to be None here. 

Something like: 
```python
if stat.language_code is None:
   raise Exception(' <<some valide message >>')
```

Reply:
Done

Reply:
```suggestion
                stat.language_code,
```

After raising error why you are still casting to str() -- Is MyPy still throwing an error ?

Reply:
```suggestion
                stat.language_code,
```

After raising the error why you are still casting to str() -- Is MyPy still throwing an error ?